src/dune_rules/coq/coq_rules.ml

```diff
@@ -448,7 +448,14 @@ let parse_line ~dir line =
     in
     (* let depname, ext = Filename.split_extension ff in *)
     let target = Path.relative (Path.build dir) target in
-    let deps = String.extract_blank_separated_words deps in
+    let deps =
+      (* Before we split the deps on spaces, we do some preprocessing to handle the case
+         where the path contains ":" and coqdep escapes this as "\:" causing Dune to
+         misinterpret the path. We simply revert that escaping alllowing this to work on
+         Windows. *)
+      Re.replace_string (Re.compile (Re.str "\\:")) ~by:":" deps
```

Comment:
Should we replace `\\` by `\` too?

Reply:
Should we enable this only on Windows?

Reply:
Maybe move the logic to its own function `unquote_make_path_on_win` ?

Reply:
I discussed replacing `\\` but decided against it to keep the patch small. It's harmless on Windows anyway.

Reply:
@ejgallego I don't think its worth making into its own function, nor worth enabling only on windows.

Reply:
I don't think it is going to be such a hotpath as we aren't parsing coqdep all the time.

Reply:
Even if we don't use this patch, we could allow the Coq platform maintainers to use it and come up with something better. At the end of the day, we are fighting hacks with hacks so making this slicker seems like dressing a pig. The more correct solution would be to have a better format for coqdep to communicate with Dune.

Reply:
I think we should use the patch; it is fine to run it unconditionally, but I think I'd like to have the code modular and have a top-level function `unescape_coqdep_path` so the different implementation concerns are separated.